secure/storage/src/tests/on_disk.rs

```diff
@@ -7,6 +7,27 @@ use libra_temppath::TempPath;
 #[test]
 fn on_disk() {
     let temp_path = TempPath::new();
-    let storage = Box::new(OnDiskStorage::new(temp_path.path().to_path_buf()));
-    suite::run_test_suite(storage, "OnDiskStorage");
+    let mut storage = Box::new(OnDiskStorage::new(temp_path.path().to_path_buf()));
+    suite::run_test_suite(storage.as_mut(), "OnDiskStorage");
+}
+
+#[test]
+fn crypto_key_create_and_get() {
+    let temp_path = TempPath::new();
+    let mut storage = Box::new(OnDiskStorage::new(temp_path.path().to_path_buf()));
+    suite::create_get_and_test_key_pair(storage.as_mut());
+}
+
+#[test]
+fn crypto_key_create_twice() {
```

Comment:
my concern is that each new storage interface will need to explicitly call tests, I want to isolate this so they are coerced to running all tests and not accidentally missing one that actually reveals an incomplete or incorrect implementation.

Reply:
Yes, agreed. I've also addressed this in my refactor PR (see my previous comment). If you're happy, we could defer this? I would address it here, but it blows up the size of the PR significantly (hence why I chose to split it!).